core/src/actors/session.rs

```diff
@@ -139,48 +204,166 @@ impl Message for GetPeers {
+                        for msg in msgs {
+                            self.send_message(msg);
                         }
-                        _ => warn!(
+                        try_consolidate_session(self, ctx);
+                    }
+                    (SessionStatus::Unconsolidated, Command::Verack) => {
+                        handshake_verack(self);
+                        try_consolidate_session(self, ctx);
+                    }
+                    ////////////////////
+                    // PEER DISCOVERY //
+                    ////////////////////
+                    (SessionStatus::Consolidated, Command::GetPeers) => {
+                        peer_discovery_get_peers(self, ctx);
+                    }
+                    (SessionStatus::Consolidated, Command::Peers { peers }) => {
+                        peer_discovery_peers(&peers);
+                    }
+                    /////////////////////
+                    // NOT IMPLEMENTED //
+                    /////////////////////
+                    (SessionStatus::Consolidated, _) => {
+                        debug!("Not implemented message command received!");
+                    }
+                    (_, kind) => {
+                        warn!(
                             "Received a message of kind \"{:?}\", which is not implemented yet",
-                            msg.kind
-                        ),
-                    },
-                }
+                            kind
+                        );
+                    }
+                };
             }
         }
     }
 }
 
-/// Handler for GetPeers message.
+/// Handler for GetPeers message (sent by other actors)
 impl Handler<GetPeers> for Session {
-    type Result = SessionGetPeersResult;
+    type Result = SessionUnitResult;
 
     fn handle(&mut self, _msg: GetPeers, _: &mut Context<Self>) {
         debug!("GetPeers message should be sent through the network");
         // Create get peers message
-        let get_peers_msg: Vec<u8> = WitnetMessage::build_get_peers().into();
+        let get_peers_msg = WitnetMessage::build_get_peers();
         // Write get peers message in session
-        self.framed.write(Response::Message(get_peers_msg.into()));
+        self.send_message(get_peers_msg);
     }
 }
 
-fn handle_get_peers_message(session: &mut Session, ctx: &mut Context<Session>) {
+////////////////////////////////////////////////////////////////////////////////////////
+// ACTOR HELPER FUNCTIONS
+////////////////////////////////////////////////////////////////////////////////////////
+/// Function called when Version message is received
+fn handshake_version(session: &mut Session, sender_address: &Address) -> Vec<WitnetMessage> {
+    info!("Version message have been received");
+    let flags = &mut session.handshake_flags;
+
+    if flags.version_rx {
+        debug!("Version message already received");
+        // TODO: placeholder to change behaviour (right only logging)
```

Comment:
What's this?

Reply:
In case that `version` or `verack` messages are received more than once, we may define strategies for punish those peers for "strange" behaviors.